tests/integration/security/sds_citadel_flow/sds_citadel_flow_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sds_citadel_test
+
+import (
+	"fmt"
+	"path"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/apps"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+var (
+	inst istio.Instance
+)
+
+type testPolicy struct {
+	t         *testing.T
+	env       *kube.Environment
+	namespace string
+	fileName  string
+}
+
+type connection struct {
+	from            apps.KubeApp
+	to              apps.KubeApp
+	protocol        apps.AppProtocol
+	port            int
+	expectedSuccess bool
+}
+
+func (p testPolicy) tearDown() {
+	p.t.Logf("Tearing down policy %q.", p.fileName)
+	if err := p.env.Delete(p.namespace, path.Join("testdata", p.fileName)); err != nil {
+		p.t.Fatalf("Cannot delete %q from namespace %q: %v", p.fileName, p.namespace, err)
+	}
+}
+
+func applyPolicyFile(t *testing.T, env *kube.Environment, namespace string, fileName string) *testPolicy {
+	t.Logf("Applying policy file %v", fileName)
+	if err := env.Apply(namespace, path.Join("testdata", fileName)); err != nil {
+		t.Fatalf("Cannot apply %q to namespace %q: %v", fileName, namespace, err)
+		return nil
+	}
+	return &testPolicy{
+		t:         t,
+		env:       env,
+		namespace: namespace,
+		fileName:  fileName,
+	}
+}
+
+func checkConnection(conn connection) error {
+	ep := conn.to.EndpointForPort(conn.port)
+	if ep == nil {
+		return fmt.Errorf("cannot get upstream endpoint for connection test %v", conn)
+	}
+
+	results, err := conn.from.Call(ep, apps.AppCallOptions{Protocol: conn.protocol})
+	if conn.expectedSuccess {
+		if err != nil || len(results) == 0 || results[0].Code != "200" {
+			// Addition log for debugging purpose.
+			if err != nil {
+				fmt.Printf("Error: %#v\n", err)
```

Comment:
You probably want to prefer `t.Log` here, rather than Printf

Reply:
Done: changed to use t.Log.